File: back/src/main/java/com/back/domain/user/dto/MyPageResponse.java
package com.back.domain.user.dto;

import com.back.domain.user.entity.User;
import java.util.List;

public record MyPageResponse(
        String name,
        String id,
        String email,
        String loginType,
        List<TicketGroupInfo> ticketGroups
) {
    public static MyPageResponse from(User user, List<TicketGroupInfo> ticketGroups) {
        return new MyPageResponse(
                user.getName(),
                user.getLoginId(),
                user.getEmail(),
                user.getLoginType().name(),
                ticketGroups
        );
    }
}

File: back/src/main/kotlin/com/back/domain/user/dto/MyPageResponse.kt
package com.back.domain.user.dto

import com.back.domain.user.entity.User
import com.back.domain.user.constant.LoginType

data class MyPageResponse(
    val name: String,
    val id: String,
    val email: String,
    val loginType: String,
    val socialProvider: String?,
    val profileImageUrl: String,
    val ticketGroups: List<TicketGroupInfo>
) {
    companion object {
        fun from(
            user: User,
            socialProvider: LoginType?,
            ticketGroups: List<TicketGroupInfo>,
        ): MyPageResponse = MyPageResponse(
            name = user.name,
            id = user.loginId,
            email = user.email,
            loginType = user.loginType.name,
            socialProvider = socialProvider?.name,
            profileImageUrl = user.redirectToProfileImgUrlOrDefault,
            ticketGroups = ticketGroups
        )
    }
}
